src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3411,12 +3416,53 @@ private PostSkipAction SkipBadAccessorListTokens(ref SyntaxToken openBrace, Synt
 
         private bool IsPossibleAccessor()
         {
-            return this.CurrentToken.Kind == SyntaxKind.IdentifierToken
+            if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken
                 || IsPossibleAttributeDeclaration()
-                || IsPossibleModifier()
                 || SyntaxFacts.GetAccessorDeclarationKind(this.CurrentToken.ContextualKind) != SyntaxKind.None
                 || this.CurrentToken.Kind == SyntaxKind.OpenBraceToken  // for accessor blocks w/ missing keyword
-                || this.CurrentToken.Kind == SyntaxKind.SemicolonToken; // for empty body accessors w/ missing keyword
+                || this.CurrentToken.Kind == SyntaxKind.SemicolonToken // for empty body accessors w/ missing keyword
+                )
+            {
+                return true;
+            }
+
+
+            return IsPossibleAccessorModifier();
```

Comment:
This should be exactly equivalent to `return (a || b || c || ... || IsPossibleAccessorModifier() );`